src/cis/fam_cis_direct.cpp

```diff
@@ -150,8 +150,14 @@ Fam_CIS_Direct::Fam_CIS_Direct(char *cisName, bool useAsyncCopy_,
             std::pair<std::string, uint64_t> service = obj->second;
             Fam_Memory_Service *memoryService = new Fam_Memory_Service_Client(
                 (service.first).c_str(), service.second);
-            memoryServers->insert({ obj->first, memoryService });
+            if (memoryServers->find(obj->first) != memoryServers->end()) {
```

Comment:
Let us move this check to get_config_info. 

After the line : 

std::vector<std::string> temp = info->get_value_list("memsrv_list");

Reply:
Sure, will make the change